frontend/src/scenes/data-warehouse/dataWarehouseOverviewLogic.ts

```diff
@@ -0,0 +1,114 @@
+import { actions, afterMount, kea, listeners, path, reducers, selectors } from 'kea'
+import { loaders } from 'kea-loaders'
+import { lemonToast } from '@posthog/lemon-ui'
+import api, { PaginatedResponse } from 'lib/api'
+
+import { ExternalDataSource } from '~/types'
+import type { dataWarehouseOverviewLogicType } from './dataWarehouseOverviewLogicType'
+
+export const dataWarehouseOverviewLogic = kea<dataWarehouseOverviewLogicType>([
+    path(['scenes', 'data-warehouse', 'dataWarehouseOverviewLogic']),
+
+    actions({
+        updateDataSource: (dataSource: ExternalDataSource) => ({ dataSource }),
+        syncDataSource: (id: string) => ({ id }),
+        resyncDataSource: (id: string) => ({ id }),
+        deleteDataSource: (id: string) => ({ id }),
+    }),
+
+    loaders({
+        dataSourcesResponse: [
+            null as PaginatedResponse<ExternalDataSource> | null,
+            {
+                loadDataSources: async () => {
+                    const response = await api.externalDataSources.list()
+                    return response
+                },
+            },
+        ],
+    }),
```

Comment:
it'd be nice to have another logic elsewhere to handle the data loading for sources, and then update all other logics that use `api.externalDataSources.list()` to use the same common logic. The reason for this is so that when the logic cache is updated in one place, UI all over the app will get updated with it - otherwise, you may end up viewing different versions of the same data on different pages